GDJS/Runtime/pixi-renderers/pixi-object-effects-manager.ts

```diff
@@ -0,0 +1,271 @@
+/*
+ * GDevelop JS Platform
+ * Copyright 2013-2021 Florian Rival (Florian.Rival@gmail.com). All rights reserved.
+ * This project is released under the MIT License.
+ */
+namespace gdjs {
+  import PIXI = GlobalPIXIModule.PIXI;
+
+  type RendererEffects = Record<string, PixiFiltersTools.Filter>;
+
+  export interface EffectsTarget {
+    getRuntimeScene: () => RuntimeScene;
+    getElapsedTime: (runtimeScene: RuntimeScene) => number;
+    getHeight: () => number;
+    getWidth: () => number; 
+  };
+
+  class PixiObjectEffectsManager {
+    /**
+     * Initialize the renderer effects from the effect data.
+     * @param effectData The effect data
+     * @param rendererEffects The renderer effects collection
+     * @param layer The layer
+     */
+    initializeEffect(
+      effectData: EffectData,
+      rendererEffects: RendererEffects,
+      target: EffectsTarget
+    ): boolean {
+      const filterCreator = gdjs.PixiFiltersTools.getFilterCreator(
+        effectData.effectType
+      );
+      if (!filterCreator) {
+        console.log(
+          'Filter "' +
+            effectData.name +
+            '" has an unknown effect type: "' +
+            effectData.effectType +
+            '". Was it registered properly? Is the effect type correct?'
+        );
+        return false;
+      }
+
+      const filter: PixiFiltersTools.Filter = {
+        pixiFilter: filterCreator.makePIXIFilter(target, effectData),
+        updateDoubleParameter: filterCreator.updateDoubleParameter,
+        updateStringParameter: filterCreator.updateStringParameter,
+        updateBooleanParameter: filterCreator.updateBooleanParameter,
+        update: filterCreator.update,
+      };
+
+      rendererEffects[effectData.name] = filter;
+      return true;
+    }
+
+    /**
+     * Apply the effect on the renderer object.
+     * @param rendererObject The renderer object
+     * @param effect The effect to be applied.
+     */
+    applyEffect(
+      rendererObject: PIXI.DisplayObject,
+      effect: PixiFiltersTools.Filter
+    ): boolean {
+      rendererObject.filters = (rendererObject.filters || []).concat(
+        effect.pixiFilter
+      );
+      return true;
+    }
+
+    /**
+     * Update the filters applied on an object.
+     * @param runtimeObject
+     * @param layer
+     */
+    update(rendererEffects: RendererEffects, layer: Layer) {
+      for (const filterName in rendererEffects) {
+        const filter = rendererEffects[filterName];
+        filter.update(filter.pixiFilter, layer);
+      }
+    }
+
+    /**
+     * Add a new effect on a runtime object, or replace the one
+     * with the same name.
+     * @param effectData The effect data
+     * @param rendererEffects The renderer effects
+     * @param rendererObject The renderer object
+     * @param layer The Layer
+     */
+    addEffect(
+      effectData: EffectData,
+      rendererEffects: RendererEffects,
+      rendererObject: PIXI.DisplayObject,
+      layer: Layer
```

Comment:
We should not pass the layer, the effects manager should just deal with `rendererObject`. Remove any mention to layer in this file.